tests/unit/Core/Framework/DataAbstractionLayer/Dbal/QueryBuilderTest.php

```diff
@@ -0,0 +1,234 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Tests\Unit\Core\Framework\DataAbstractionLayer\Dbal;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Shopware\Core\Framework\DataAbstractionLayer\Dbal\QueryBuilder;
+
+/**
+ * @internal
+ */
+#[CoversClass(QueryBuilder::class)]
+class QueryBuilderTest extends TestCase
+{
+    private Connection&MockObject $connection;
+    private QueryBuilder $queryBuilder;
+
+    protected function setUp(): void
+    {
+        $platform = $this->createMock(AbstractPlatform::class);
+
+        $this->connection = $this->createMock(Connection::class);
+        $this->connection->method('getDatabasePlatform')->willReturn($platform);
+
+        $this->queryBuilder = new QueryBuilder($this->connection);
+    }
+
+    public function testSetAndGetComment(): void
+    {
+        // Initial value should be null
+        static::assertNull($this->queryBuilder->getComment());
+
+        // Test setting a comment
+        $this->queryBuilder->setComment('TEST_COMMENT');
+        static::assertSame('TEST_COMMENT', $this->queryBuilder->getComment());
+
+        // Test setting null
+        $this->queryBuilder->setComment(null);
+        static::assertNull($this->queryBuilder->getComment());
+    }
+
+    public function testGetSQLWithComment(): void
+    {
+        // We'll simulate what getSQL() does but in a controlled way
+        // First set up our QueryBuilder
+        $queryBuilder = new QueryBuilder($this->connection);
+        
+        // Set a comment
+        $queryBuilder->setComment('+ MAX_EXECUTION_TIME(1000)');
+        
+        // Create a reflection class to access private methods/properties
+        $reflectionClass = new \ReflectionClass(QueryBuilder::class);
+        $commentProperty = $reflectionClass->getProperty('comment');
+        $commentProperty->setAccessible(true);
+        
+        // Verify the comment is stored correctly
+        static::assertEquals('+ MAX_EXECUTION_TIME(1000)', $commentProperty->getValue($queryBuilder));
```

Comment:
```suggestion
        static::assertSame('+ MAX_EXECUTION_TIME(1000)', $commentProperty->getValue($queryBuilder));
```

please fix all other occurrences